Revit_Core_Engine/Query/DrawingArea.cs

```diff
@@ -0,0 +1,98 @@
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.Engine.Geometry;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Compute the outline of Title Block drawing area.")]
+        [Input("titleBlock", "Title Block symbol to get the drawing area outline from.")]
+        [Output("outline", "The Title Block's drawing area.")]
+        public static Outline DrawingArea(this FamilySymbol titleBlock)
+        {
+            Document document = titleBlock.Document;
+            Document familyDoc = document.EditFamily(titleBlock.Family);
+            List<DetailLine> lines = new FilteredElementCollector(familyDoc).OfCategory(Autodesk.Revit.DB.BuiltInCategory.OST_Lines).WhereElementIsNotElementType().Where(x => x is DetailLine).Cast<DetailLine>().ToList();
+
+            var compositeGeom = new CompositeGeometry();
+
+            foreach (DetailLine dLine in lines)
+            {
+                var bhomLine = dLine.GeometryCurve.IFromRevit() as BH.oM.Geometry.Line;
+
+                if (bhomLine != null)
+                    compositeGeom.Elements.Add(bhomLine);
+            }
+
+            BH.oM.Geometry.Point centrePoint = compositeGeom.Bounds().Centre();
+            
+            var leftLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint - Vector.XAxis * 10);
+            var rightLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint + Vector.XAxis * 10);
+            var upLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint + Vector.YAxis * 10);
+            var downLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint - Vector.YAxis * 10);
+
+            double minX = double.MinValue;
```

Comment:
2nd thought on that - I would consider using other variable name, e.g. `leftBound` etc. Because in fact we are not looking for minimum X, rather max X to the left from centre point. Sorry for the mess